Document/0x05c-Reverse-Engineering-and-Tampering.md

```diff
@@ -290,6 +290,95 @@ For white-box source code testing, you'll need a setup similar to the developer'
+...
+Hello from C++
+...
+```
+
+The output can also be piped to a text file, and using other Unix utilities like `grep`, `head` etc for further analysis. 
+
+##### Cross References
+
+###### Java
+
+Java cross references is supported by many tools, including Ghidra and [jadx](https://github.com/skylot/jadx "jadx"). With Ghidra cross references can be obtained by right clicking on the desired function in the symbol tree and selecting **Show References to**, as shown in the image below.
+
+<img src="Images/Chapters/0x05c/Ghidra_java_xrefs.png" width="500px"/>
+
+Another popular Java decompiler tool is jadx. Using jadx, cross references for a desired function can be obtained by right clicking the decompiled function and selecting **Find Usage**.
+
+<img src="Images/Chapters/0x05c/jadx_java_xrefs.png" width="500px"/>
+
+
+###### Native
+
+While analysing an Android application's native library using Ghidra, cross references can be obtained similarly to Java, i.e, by right clicking the desired function and selecting **Show References to**.
+
+<img src="Images/Chapters/0x05c/ghidra_native_xrefs.png" width="500px"/>
+
+
+##### API Usage
+
+The Android platform provides many in-built libraries for frequently used functionalities in applications, for example cryptography, Bluetooth, NFC, network or location libraries. Presence of these imported libraries in an application's code can give us valuable information about the nature of application being evaluated.
+
+For instance, if an application is importing `javax.crypto.Cipher`, it indicates that the application will be performing some kind of cryptographic operation. Fortunately, Cryptographic calls are very standard in nature, i.e, they need to be called in a particular order to work correctly, this knowledge can be helpful when analysing cryptography APIs. For example, by looking for the `Cipher.getInstance()` function, we can determine the cryptographic algorithm being used. With such an approach we can directly move to analysing cryptographic assets, which often are very critical in an application. Further information on how to analyse Android's cryptographic APIs is discussed in the section  "[Android Cryptographic APIs](0x05e-testing-cryptography "Android Cryptographic APIs")".
+
+Similarly, the above approach can be used in determining where and how an application is using NFC. For instance, an application using Host-based Card Emulation for performing digital payments, must use the `android.nfc` package. By searching for critical functions like `processCommandApdu()` from the `android.nfc.cardemulation.HostApduService` class, this information can provide us a starting point for NFC API analysis.
+
+Checking for the imported classes in an application can help us in understanding the functionalities implemented. Further by determining the most important functions of an imported class, one can start analysis by assuming this function as the centrepiece of the puzzle.
+
```

Comment:
I think it's already well explained with the intro and the examples. We don't really need a closing paragraph here.
```suggestion
```